### PR TITLE
chore: add kubebuilder RBAC markers

### PR DESCRIPTION
**Pull request by MatousJobanek**
add RBAC kubebuilder markers and regenerate the cluster role to test that all needed permissions are granted.
In the future, the cluster role will be generated on-the-fly together with creating an operator bundle.

The PR also add the `get-tool.mk` file that downloads the controller-gen binary.

is paired with https://github.com/codeready-toolchain/toolchain-e2e/pull/313

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]
Looks good but I think the nstemplateset controller has wrong type

File: controllers/nstemplateset/nstemplateset_controller.go [hunk 1]
```diff
@@ -90,6 +90,15 @@ type Reconciler struct {
 	status           *statusManager
 }
 
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=idlers,verbs=get;list;watch;create;update;patch;delete
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply controllers/nstemplateset/nstemplateset_controller.go]: good catch, thanks :+1: 
fixed in: 06eb0b5

**Review by xcoulon — APPROVED** [review]

**Pull request merged**